sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessCreateTest.java

```diff
@@ -196,4 +211,17 @@ public void testGetDefaultOutputCoderReturnsConstructorCoder() throws Exception
     Coder<Integer> defaultCoder = source.getDefaultOutputCoder();
     assertThat(defaultCoder, equalTo(coder));
   }
+
+  @Test
+  public void testSplitAtFraction() throws Exception {
+    List<Integer> elements = new ArrayList<>();
+    Random random = new Random();
+    for (int i = 0; i < 50; i++) {
+      elements.add(random.nextInt());
+    }
+    InProcessCreate.InMemorySource<Integer> source =
+        new InMemorySource<>(elements, VarIntCoder.of());
+
+    SourceTestUtils.assertSplitAtFractionExhaustive(source, PipelineOptionsFactory.create());
```

Comment:
On my machine this is slightly less than 1 second. Reduced to 25, which is slightly more than 0.5s, but should still be sufficiently interesting.